internal/manifests/manifestutils/dns.go

```diff
@@ -15,14 +15,22 @@
 package manifestutils
 
 import (
+	"errors"
+
 	corev1 "k8s.io/api/core/v1"
 )
 
+var ErrorDNSPolicy = errors.New("when a dnsPolicy is set to None, the dnsConfig field has to be specified")
+
 // Get the Pod DNS Policy depending on whether we're using a host network.
-func GetDNSPolicy(hostNetwork bool) corev1.DNSPolicy {
+func GetDNSPolicy(hostNetwork bool, dnsConfig corev1.PodDNSConfig) corev1.DNSPolicy {
 	dnsPolicy := corev1.DNSClusterFirst
 	if hostNetwork {
 		dnsPolicy = corev1.DNSClusterFirstWithHostNet
 	}
+	// If local DNS configuration is set, takes precedence of hostNetwork.
+	if dnsConfig.Nameservers != nil {
+		dnsPolicy = corev1.DNSNone
```

Comment:
it's also possible that someone would want to use nameservers but not have the DNSPolicy set to be None https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-dns-config though yeah this probably doesn't make a ton of sense... let's do this for now and if someone really wants the customization we can cross that bridge when we get to it.